frontend/src/components/blocks/BlockDetails.tsx

```diff
@@ -82,9 +82,16 @@ const BlockDetails = ({ block }: Props) => {
                     }
                     imgLink="/static/images/icon-t-status.svg"
                     text={
-                      typeof context.finalTimestamp === "undefined"
+                      context.finalityStatus &&
+                      typeof context.finalityStatus
+                        .finalBlockTimestampNanosecond === "undefined"
                         ? "Checking Finality..."
-                        : new BN(block.timestamp).lte(context.finalTimestamp)
+                        : context.finalityStatus &&
```

Comment:
You don't need to check for `context.finalityStatus` here since it was checked above